addons/html_editor/static/src/main/font/font_plugin.js

```diff
@@ -305,4 +307,36 @@ export class FontPlugin extends Plugin {
         this.shared.setCursorStart(p);
         return true;
     }
+
+    onInput(ev) {
+        if (ev.data !== " ") {
+            return;
+        }
+        const selection = this.shared.getEditableSelection();
+        const blockEl = closestBlock(selection.anchorNode);
+        const leftDOMPath = leftLeafOnlyNotBlockPath(selection.anchorNode);
+        let spaceOffset = selection.anchorOffset;
+        let leftLeaf = leftDOMPath.next().value;
+        while (leftLeaf) {
+            // Calculate spaceOffset by adding lengths of previous text nodes
+            // to correctly find offset position for selection within inline
+            // elements. e.g. <p>ab<strong>cd []e</strong></p>
+            spaceOffset += leftLeaf.length;
+            leftLeaf = leftDOMPath.next().value;
+        }
+        const precedingText = blockEl.textContent.substring(0, spaceOffset);
```

Comment:
Here we need the actual position of cursor from starting of text to get the preceding text from cursor. If there is a simple text like this: `<p># ab[]c</p>` we can get the preceding text by selection.anchorOffset. But it won't work in such case: `<p># ab<span>c[]d</span></p>`. In this case if user hits space then selection.anchorOffset will be 2 and `precedingText` would be `# ` which will be true the condition and case will be failed. To get the actual cursor position for such cases we are iterating a loop until the most left text node from current position. So in this case `<p># ab<span>c[]d</span></p>` offset will be 6 and `presedingText` actual text will be `# abc ` .

Reply:
@adsa-odoo But it doesn't matter where the cursor is. The only thing that matters is the fact that the block matches the regex `/^#{1,6} /`.

Reply:
In your example `<p>#a[]b</p>` won't match the regex `/^#{1,6} /` because the space is not directly following the `#` so it will work.

Reply:
Yes, it does. But if I take `blockEl.textContent` and not `precedingText` to check the regex `/^#{1,6} /` then `<p>#[]ab</p>` will be failed too.

Reply:
I don't understand what you mean. If the text content is `#ab` then no heading should be created in that case so it is correct that the regex fails.

Reply:
If the text content is `#ab` then there are 2 cases:
1) user puts cursor just after the # and press space like this `<p>#[]ab</p>`
2) user puts cursor somewhere in middle of the text like this `<p>#a[]b</p>`

In first case, heading should be created when in second case heading should NOT be created.

Reply:
The case you mentioned will indeed fail if we are matching `textContent` to the regex `/^#{1,6} /`. According to your suggestion we can do something like this:
```suggestion
        const textContent = blockEl.textContent;
        const matchedPattern = textContent.match(/^(#{1,6})\s/);
        if (matchedPattern && matchedPattern.index + matchedPattern[0].length === spaceOffset) {
            const numberOfHash = matchedPattern.length - 1;
       
```
But even in this approach we will need to get correct space offset. If we are directly check `selection.anchorOffset` then this case `<p># a<strong>b []c</strong></p>` will be failed because here offset will be 2 and condition will be true.

Reply:
You're right, it doesn't work either. I suggest to use the same mechanism as the other similar features which are "1." and stuff like that:
```js
const selection = this.shared.getEditableSelection();
const blockEl = closestBlock(selection.anchorNode);
const stringToConvert = blockEl.textContent.substring(0, selection.anchorOffset);
const shouldCreateHeading = /^#{1,6} /.test(stringToConvert);
```

Reply:
I am already using the same mechanism for headings as suggested by you and it is working fine. But if we remove the calculation of `spaceOffset` and use `selection.anchorOffset` then it is not working for this case `<p># a<strong>b []c</strong></p>`.

Reply:
> I suggest to use the same mechanism as the other similar features which are "1." and stuff like that:

@dmo-odoo Here @adsa-odoo is referring to this test case https://github.com/odoo/odoo/pull/175991/files#diff-d81d0af6a68236af6b847f83b5de31939e701aaaddad1dd35bfd0cc16179bfeeR543 also the same issue is with similar feature like "1." which we fix in commit 2.

Reply:
Oh ok, I'm sorry then. I thought the code should be the same between those similar features, but I didn't understand that the original one had the same issue. Thank you!